securesystemslib/signer/_gcp_signer.py

```diff
@@ -180,7 +180,7 @@ def _get_hash_algorithm(public_key: Key) -> str:
             )
 
         # trigger UnsupportedAlgorithm if appropriate
```

Comment:
This is now a ValueError, which breaks the API and is also inconsistent with above, where we raise an UnsupportedAlgorithmError for some unsupported algorithms. Same is true for other signers in this PR

I'll mark the PR as draft until I fix this.

Reply:
Turns out, other signers exhaustively check public key schemes prior to using the hash algorithm name. This menas `UnsupportedAlgorithmError` is not expected when a digest object is created in those signers.

d5016ae adds a similar scheme check to GCPSigner to make use of the already existing `raise UnsupportedAlgorithmError`, previously only used for invalid key types.

 f5f162f proposes an alternative, less invasive but IMO more clunky fix, which just re-raises hashlib's exception as `UnsupportedAlgorithmError`. 

Either way, we may want to consider a more comprehensive solution in #593 and #766.